### PR TITLE
Migrate profiles controller tests to rspec

### PR DESCRIPTION
**Pull request by OmTheTurtle**

### CODE REVIEW

**Comment by falusi94** [comment]
Here is a good [explanation from DHH](https://github.com/rails/rails/issues/18950) on my problem. Checking these instance variables means that the test knows more about the underlying application code than it should. We are on request/controller level (almost like a browser), so we shouldn't be aware of these variables, but the data in the response.

Although on the other hand I feel your need on testing somehow the validity of the returned data. I can imagine checking the content of the response body for relevant info.

File: spec/requests/profiles_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+describe ProfilesController do
+  before do
+    @sanyi = create(:user)
+    @bela  = create(:user)
+    login_as(@sanyi)
+  end
```
[the remark this answers is not in the document]
Comment (OmTheTurtle) [reply spec/requests/profiles_controller_spec.rb]: It's better, and works, I'll refactor.

**Comment by OmTheTurtle** [comment]
I was looking into the mentioned problem and decided to not care about this right now. Testing the correct status code and redirects are going to be enough for now.

File: spec/requests/profiles_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      it 'redirects to own profile' do
+        get '/profiles/dummy'
+
+        expect(response).to redirect_to(profiles_me_path)
+      end
+    end
+  end
+
+  describe '#show' do
+    it 'shows the own profile of the user' do
```
Comment (falusi94) [inline spec/requests/profiles_controller_spec.rb]: I think you wanted to put this to the description of the `#show_self` and to here without the ‘own’.

**Pull request merged**